services/motion/builtin/builtin_test.go

```diff
@@ -1446,22 +1141,888 @@ func TestMoveOnGlobe(t *testing.T) {
+		easyGoalInBaseFrame := spatialmath.NewPoseFromPoint(r3.Vector{X: 0.277 * 1000, Y: 0.593 * 1000})
+		easyGoalInSLAMFrame := spatialmath.PoseBetweenInverse(motion.SLAMOrientationAdjustment, easyGoalInBaseFrame)
+
+		req := motion.MoveOnMapReq{
+			ComponentName: base.Named("test-base"),
+			Destination:   easyGoalInSLAMFrame,
+			SlamName:      slam.Named("test_slam"),
+			Extra:         map[string]interface{}{"smooth_iter": 0},
+		}
+
+		executionID, err := ms.(*builtIn).MoveOnMapNew(context.Background(), req)
+		test.That(t, err, test.ShouldBeNil)
+		test.That(t, executionID, test.ShouldNotBeEmpty)
+
+		// returns the execution just created in the history
+		ph, err := ms.PlanHistory(ctx, motion.PlanHistoryReq{ComponentName: req.ComponentName})
+		test.That(t, err, test.ShouldBeNil)
+		test.That(t, len(ph), test.ShouldEqual, 1)
+		test.That(t, ph[0].Plan.ExecutionID, test.ShouldResemble, executionID)
+		test.That(t, len(ph[0].StatusHistory), test.ShouldEqual, 1)
+		test.That(t, ph[0].StatusHistory[0].State, test.ShouldEqual, motion.PlanStateInProgress)
+		test.That(t, len(ph[0].Plan.Steps), test.ShouldNotEqual, 0)
+
+		err = ms.StopPlan(ctx, motion.StopPlanReq{ComponentName: kb.Name()})
+		test.That(t, err, test.ShouldBeNil)
+
+		ph2, err := ms.PlanHistory(ctx, motion.PlanHistoryReq{ComponentName: req.ComponentName})
+		test.That(t, err, test.ShouldBeNil)
+		test.That(t, len(ph2), test.ShouldEqual, 1)
+		test.That(t, ph2[0].Plan.ExecutionID, test.ShouldResemble, executionID)
+		test.That(t, len(ph2[0].StatusHistory), test.ShouldEqual, 2)
+		test.That(t, ph2[0].StatusHistory[0].State, test.ShouldEqual, motion.PlanStateStopped)
+		test.That(t, ph2[0].StatusHistory[1].State, test.ShouldEqual, motion.PlanStateInProgress)
+		test.That(t, len(ph2[0].Plan.Steps), test.ShouldNotEqual, 0)
+
+		// Proves that calling StopPlan after the plan has reached a terminal state is idempotent
+		err = ms.StopPlan(ctx, motion.StopPlanReq{ComponentName: kb.Name()})
+		test.That(t, err, test.ShouldBeNil)
+		ph3, err := ms.PlanHistory(ctx, motion.PlanHistoryReq{ComponentName: req.ComponentName})
+		test.That(t, err, test.ShouldBeNil)
+		test.That(t, ph3, test.ShouldResemble, ph2)
+	})
+
+	t.Run("Is able to reach the zero pose with an empty motion configuration", func(t *testing.T) {
+		_, ms := createMoveOnMapEnvironment(ctx, t, "pointcloud/octagonspace.pcd", 40, nil)
+		defer ms.Close(ctx)
+
+		req := motion.MoveOnMapReq{
+			ComponentName: base.Named("test-base"),
+			Destination:   spatialmath.NewZeroPose(),
+			SlamName:      slam.Named("test_slam"),
+			MotionCfg:     &motion.MotionConfiguration{},
+		}
+
+		executionID, err := ms.(*builtIn).MoveOnMapNew(context.Background(), req)
+		test.That(t, err, test.ShouldBeNil)
+		test.That(t, executionID, test.ShouldNotResemble, uuid.Nil)
+	})
+}
+
+func TestMoveCallInputs(t *testing.T) {
+	t.Parallel()
+	ctx := context.Background()
+
+	t.Run("Returns error when called with an unknown component", func(t *testing.T) {
```

Comment:
Could we please group the MoveOnGlobe & MoveOnGlobe inputs into sub t.Runs?


```suggestion
        t.Run("MoveOnMapNew", , func(t *testing.T) {
	    t.Run("Returns error when called with an unknown component", func(t *testing.T) {
```